gegelatilib/src/learn/learningAgent.cpp

```diff
@@ -368,3 +368,9 @@ void Learn::LearningAgent::keepBestPolicy()
         }
     }
 }
+
+void Learn::LearningAgent::forgetPreviousResults()
+{
+    resultsPerRoot.clear();
+    bestRoot.second = nullptr;
```

Comment:
Shouldn't bestRoot.first be reset too?
```suggestion
    bestRoot.first = nullptr;
```
Without this, I'm afraid a segmentation fault my occur in LearningAgent.cpp:335
```
  if (this->bestRoot.first == nullptr         // NULL case
            || *this->bestRoot.second < *evaluation 
```
If the `bestRoot.first` is not null, a comparison may occur with the nullptr of `bestRoot.second`.